pkg/apiserver/debugapi/endpoint_def.go

```diff
@@ -0,0 +1,539 @@
+		},
+	},
+	{
+		ID:        "tidb_config",
+		Component: model.NodeKindTiDB,
+		Path:      "/settings",
+		Method:    endpoint.MethodGet,
+	},
+	{
+		ID:        "tidb_schema",
+		Component: model.NodeKindTiDB,
+		Path:      "/schema",
+		Method:    endpoint.MethodGet,
+		QueryParams: []*endpoint.APIParam{
+			{Model: APIParamModelTableID, Name: "table_id"},
+		},
+	},
+	{
+		ID:        "tidb_schema_db",
+		Component: model.NodeKindTiDB,
+		Path:      "/schema/{db}",
+		Method:    endpoint.MethodGet,
+		PathParams: []*endpoint.APIParam{
+			{Model: APIParamModelDB, Name: "db", Required: true},
+		},
+	},
+	{
+		ID:        "tidb_schema_db_table",
+		Component: model.NodeKindTiDB,
+		Path:      "/schema/{db}/{table}",
+		Method:    endpoint.MethodGet,
+		PathParams: []*endpoint.APIParam{
+			{Model: APIParamModelDB, Name: "db", Required: true},
+			{Model: APIParamModelTable, Name: "table", Required: true},
+		},
+	},
+	{
+		ID:        "tidb_dbtable_tableid",
+		Component: model.NodeKindTiDB,
+		Path:      "/db-table/{tableID}",
+		Method:    endpoint.MethodGet,
+		PathParams: []*endpoint.APIParam{
+			{Model: APIParamModelTableID, Name: "tableID", Required: true},
+		},
+	},
+	{
+		ID:        "tidb_ddl_history",
+		Component: model.NodeKindTiDB,
+		Path:      "/ddl/history",
+		Method:    endpoint.MethodGet,
+	},
+	{
+		ID:        "tidb_info",
+		Component: model.NodeKindTiDB,
+		Path:      "/info",
+		Method:    endpoint.MethodGet,
+	},
+	{
+		ID:        "tidb_info_all",
+		Component: model.NodeKindTiDB,
+		Path:      "/info/all",
+		Method:    endpoint.MethodGet,
+	},
+	{
+		ID:        "tidb_regions_meta",
+		Component: model.NodeKindTiDB,
+		Path:      "/regions/meta",
+		Method:    endpoint.MethodGet,
+	},
+	{
+		ID:        "tidb_region_id",
+		Component: model.NodeKindTiDB,
+		Path:      "/regions/{regionID}",
+		Method:    endpoint.MethodGet,
+		PathParams: []*endpoint.APIParam{
+			{Model: APIParamModelText, Name: "regionID", Required: true},
+		},
+	},
+	{
+		ID:        "tidb_table_regions",
+		Component: model.NodeKindTiDB,
+		Path:      "/tables/{db}/{table}/regions",
+		Method:    endpoint.MethodGet,
+		PathParams: []*endpoint.APIParam{
+			{Model: APIParamModelDB, Name: "db", Required: true},
+			{Model: APIParamModelTable, Name: "table", Required: true},
+		},
+	},
+	{
+		ID:        "tidb_hot_regions",
+		Component: model.NodeKindTiDB,
+		Path:      "/regions/hot",
+		Method:    endpoint.MethodGet,
+	},
+	{
+		ID:        "tidb_pprof",
+		Component: model.NodeKindTiDB,
+		Path:      "/debug/pprof/{kind}",
+		Method:    endpoint.MethodGet,
+		PathParams: []*endpoint.APIParam{
+			pprofKindsParam,
+		},
+		QueryParams: []*endpoint.APIParam{
+			pprofSecondsParam,
+			pprofDebugParam,
+		},
+		OnResolve: func(resolvedPayload *endpoint.ResolvedRequestPayload) error {
+			if err := timeoutMiddleware(resolvedPayload, resolvedPayload.QueryParams.Get("seconds")); err != nil {
+				return err
+			}
+			return nil
+		},
```

Comment:
this can be `OnResolve: resolveTimeoutFromQuery("seconds")`

Reply:
Master!